s3transfer/download.py

```diff
@@ -614,6 +614,38 @@ def _main(self):
         pass
 
 
+class DownloadChunkIterator(object):
+    def __init__(self, body, chunksize, response):
+        """Iterator to chunk out a downloaded S3 stream
+
+        :param body: A readable file-like object
+        :param chunksize: The amount to read each time
+        :param response: A S3 GetObject response
+        """
+        self._body = body
+        self._chunksize = chunksize
+        self._response = response
+        self._num_reads = 0
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        chunk = self._body.read(self._chunksize)
+        self._num_reads += 1
+        if chunk:
+            return chunk
+        elif self._num_reads == 1 and not self._response.get('ContentLength'):
```

Comment:
Now that I think about it is not actually needed.  I originally had ugly logic where I tried to handle a lot of this logic in the ``GetObjectTask`` based off the response, but then I decided to create a new iterator for it, but since we are able to track reads we do not need to check ``ContentLength`` anymore. I will update the class.